lib/htmlReporter.js

```diff
@@ -0,0 +1,73 @@
+const path = require("path");
+const fs = require("fs");
+const http = require("http");
+const chalk = require("chalk");
+const mkdir = require("mkdirp");
+
+const ejs = require("ejs");
+
+const projectRoot = path.resolve(__dirname, "..");
+
+const defaultOptions = {
+  reportFileDir: "public",
+  appFileName: "reporter.js",
+  reporterFileName: "report.html"
+};
+
+function getAssetsContent(filename) {
+  return fs.readFileSync(`${projectRoot}/public/${filename}`, "utf-8");
+}
+
+class HTMLReporter {
+  constructor(duplicates, opts) {
+    this.duplicates = duplicates;
+    this.opts = Object.assign({}, defaultOptions, opts);
+
+    this.renderFile = this.renderFile.bind(this);
+    this.printError = this.printError.bind(this);
+  }
+
+  printError(err) {
+    process.stderr.write(chalk.red(err));
+  }
+
+  renderFile() {
+    if (!this.duplicates) {
+      return;
+    }
+
+    ejs.renderFile(
+      `${projectRoot}/views/index.ejs`,
+      {
+        mode: "static",
+        appFileName: this.opts.appFileName,
+        dumpAssets: getAssetsContent,
+        duplicates: JSON.stringify({
+          data: this.duplicates.map(duplicate => {
```

Comment:
I think I should be able to read the whole file from the `path` and add it as part of this object. The only thing I am worried it the size the JSON we embed into the HTML will increase very quickly based on the size of the file.

But since this is a static file, I don't think this should be a problem. What do you think?